src/telemetry/client.ts

```diff
@@ -0,0 +1,144 @@
+/**
+ * PostHog Telemetry Client
+ *
+ * Provides a type-safe interface for sending telemetry events to PostHog.
+ * All payloads are defined in ./payload.ts for transparency.
+ */
+
+import posthog from "posthog-js";
+import type { TelemetryEventPayload } from "./payload";
+
+// Default configuration
+const DEFAULT_POSTHOG_KEY = "phc_vF1bLfiD5MXEJkxojjsmV5wgpLffp678yhJd3w9Sl4G";
+const DEFAULT_POSTHOG_HOST = "https://us.i.posthog.com";
+
+// Environment variables with fallback to public defaults
+function getPosthogConfig(): { key: string; host: string } {
+  // In browser/Vite environment, use import.meta.env
+  // In test/Node environment, fall back to defaults
+  try {
+    // Use eval to avoid TypeScript compile-time checking
+    // eslint-disable-next-line @typescript-eslint/no-implied-eval, @typescript-eslint/no-unsafe-call
+    const importMeta = new Function("return import.meta")() as
+      | { env?: { VITE_PUBLIC_POSTHOG_KEY?: string; VITE_PUBLIC_POSTHOG_HOST?: string } }
+      | undefined;
+    if (importMeta?.env) {
+      return {
+        // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+        key: importMeta.env.VITE_PUBLIC_POSTHOG_KEY || DEFAULT_POSTHOG_KEY,
+        // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+        host: importMeta.env.VITE_PUBLIC_POSTHOG_HOST || DEFAULT_POSTHOG_HOST,
+      };
+    }
+  } catch {
+    // import.meta not available (Node/test environment)
+  }
+
+  return {
+    key: DEFAULT_POSTHOG_KEY,
+    host: DEFAULT_POSTHOG_HOST,
+  };
+}
+
+let isInitialized = false;
+let isTesting = false;
```

Comment:
this particular global state seems unnecessary -- the isTestEnvironment is fast enough